extensions/interactions/TextInput/protractor.js

```diff
@@ -27,12 +27,55 @@ var customizeInteraction = function(elem, placeholderText, heightOfBox) {
   ).setValue(heightOfBox);
 };
 
-var expectInteractionDetailsToMatch = function(elem) {
-  // TODO (Jacob) add checks for the placeholder text and the height of the
-  // textbox.
+var expectInteractionDetailsToMatch = function(elem, placeholderText,
+  heightOfBox) {
+  var textInputElem = elem.element(by.tagName('oppia-interactive-text-input'));
   expect(
-    elem.element(by.tagName('oppia-interactive-text-input')).isPresent()
+    textInputElem.isPresent()
   ).toBe(true);
+
+  var isCheckForPlaceHolder = (typeof placeholderText !== 'undefined');
+  var isCheckForBoxHeight = (typeof heightOfBox !== 'undefined');
+
+  if (isCheckForPlaceHolder || isCheckForBoxHeight) {
+    var textAreaElem = textInputElem.element(by.tagName('textarea'));
```

Comment:
Our preferred way to refer to elements in the DOM is by adding "protractor-test-....." classes to them and then using by.css to access those classes. Please see https://github.com/oppia/oppia/wiki/Writing-End-to-End-Tests#writing-utilities for more details.